azurecompute/src/main/java/org/jclouds/azurecompute/domain/DataVirtualHardDiskParam.java

```diff
@@ -0,0 +1,156 @@
+   @Nullable public abstract Integer logicalDiskSizeInGB();
+
+   /**
+    * If the disk that is being added is already registered in the subscription or the VHD for the disk already exists
+    * in blob storage, this element is ignored. If a VHD file does not exist in blob storage, this element defines the
+    * location of the new VHD that is created when the new disk is added.
+    * Example:
+    * http://example.blob.core.windows.net/disks/mydatadisk.vhd
+    */
+   @Nullable public abstract URI mediaLink();
+
+   /**
+    * If the disk that is being added is already registered in the subscription or the VHD for the disk does not exist
+    * in blob storage, this element is ignored. If the VHD file exists in blob storage, this element defines the path to
+    * the VHD and a disk is registered from it and attached to the virtual machine.
+    */
+   @Nullable public abstract URI sourceMediaLink();
+
+   public Builder toBuilder() {
+      return builder().fromDataVirtualHardDiskParams(this);
+   }
+
+   public static Builder builder() {
+      return new Builder();
+   }
+
+   public static final class Builder {
+
+      private String hostCaching;
+      private String diskLabel;
+      private String diskName;
+      private Integer LUN;
+      private Integer logicalDiskSizeInGB;
+      private URI mediaLink;
+      private URI sourceMediaLink;
+
+      public Builder hostCaching(String hostCaching) {
+         this.hostCaching = hostCaching;
+         return this;
+      }
+
+      public Builder diskLabel(String diskLabel) {
+         this.diskLabel = diskLabel;
+         return this;
+      }
+
+      public Builder diskName(String diskName) {
+         this.diskName = diskName;
+         return this;
+      }
+
+      public Builder LUN(Integer LUN) {
+         this.LUN = LUN;
+         return this;
+      }
+
+      public Builder logicalDiskSizeInGB(Integer logicalDiskSizeInGB) {
+         this.logicalDiskSizeInGB = logicalDiskSizeInGB;
+         return this;
+      }
+
+      public Builder mediaLink(URI mediaLink) {
+         this.mediaLink = mediaLink;
+         return this;
+      }
+
+      public Builder sourceMediaLink(URI sourceMediaLink) {
+         this.sourceMediaLink = sourceMediaLink;
+         return this;
+      }
+
+      public Builder fromDataVirtualHardDiskParams(DataVirtualHardDiskParam in) {
+         return hostCaching(in.hostCaching()).diskLabel(in.diskLabel()).diskName(in.diskName()).LUN(in.LUN())
+               .logicalDiskSizeInGB(in.logicalDiskSizeInGB()).mediaLink(in.mediaLink())
+               .sourceMediaLink(in.sourceMediaLink());
+      }
+
+   }
+
+   public static DataVirtualHardDiskParam create(String hostCaching, String diskLabel, String diskName, Integer lun,
+         Integer logicalDiskSizeInGB, URI mediaLink, URI sourceMediaLink) {
+      return new AutoValue_DataVirtualHardDiskParam(hostCaching, diskLabel, diskName, lun, logicalDiskSizeInGB,
+            mediaLink, sourceMediaLink);
```

Comment:
All values are nullable? When is it possible to have such an object with no values?